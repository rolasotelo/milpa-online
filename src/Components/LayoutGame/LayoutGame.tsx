import React, { ReactNode } from "react";
import { useHistory } from "react-router-dom";

interface Props {
  children: ReactNode;
  players: { local: string | undefined; remote: string };
  yourTurn: boolean;
}

const LayoutGame = (props: Props) => {
  const h = useHistory();
  return (
    <div className="bg-mexicanBone min-h-screen">
      <div className="bg-mexicanBlue">
        <div className="container mx-auto p-2">
          <nav className="flex justify-between text-mexicanBone">
            <div
              className="self-center w-5/12"
              style={{
                fontFamily: "goodlife-sans-condensed, sans-serif",
                fontWeight: 400,
                fontStyle: "normal",
                fontSize: "2rem",
              }}
            >
              <a href="#">{`${props.players.local?.toUpperCase()} : 0 üç´`}</a>
              {props.yourTurn && (
                <a
                  href="#"
                  className="text-mexicanPink ml-6"
                >{`<== YOUR TURN`}</a>
              )}
            </div>
            <div>
              <div
                className="self-center w-2/12 text-mexicanPink"
                style={{
                  fontFamily: "goodlife-sans-condensed, sans-serif",
                  fontWeight: 400,
                  fontStyle: "normal",
                  fontSize: "2rem",
                  textAlign: "center",
                }}
              >
                <a href="#">VS</a>
              </div>
            </div>
            <div
              className="self-center w-5/12"
              style={{
                fontFamily: "goodlife-sans-condensed, sans-serif",
                fontWeight: 400,
                fontStyle: "normal",
                fontSize: "2rem",
                textAlign: "right",
              }}
            >
              {!props.yourTurn && props.players.remote && (
                <a
                  href="#"
                  className="text-mexicanPink mr-6"
                >{`YOUR TURN ==>`}</a>
              )}
              <a href="#">
                {props.players.remote
                  ? `0 üç´ : ${props.players.remote?.toUpperCase()}`
                  : "Waiting..."}
              </a>
            </div>
          </nav>
        </div>
      </div>
      <div className="max-w-7xl flex flex-col justify-center py-4 mx-auto">
        {props.children}
      </div>
    </div>
  );
};

export default LayoutGame;
