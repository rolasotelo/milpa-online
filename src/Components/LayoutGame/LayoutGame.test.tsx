import React from "react";
import renderer from "react-test-renderer";

import LayoutGame from "./LayoutGame";

describe("<LayoutGame />", () => {
  const defaultProps = {
    children: <div>Hola</div>,
  };
  const wrapper = renderer.create(<LayoutGame {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
