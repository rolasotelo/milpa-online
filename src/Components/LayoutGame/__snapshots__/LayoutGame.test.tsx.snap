// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutGame /> render 1`] = `
<div
  className="bg-mexicanBone min-h-screen"
>
  <div
    className="bg-mexicanBlue"
  >
    <div
      className="container mx-auto p-2"
    >
      <nav
        className="flex justify-between text-mexicanPink"
      >
        <div
          className="self-center"
        >
          <a
            href="#"
          >
            Menu
          </a>
        </div>
        <div>
          <button
            className="bg-button-logo focus:bg-button-logo-pressed focus:outline-none"
            onClick={[Function]}
            style={
              Object {
                "height": "50px",
                "width": "50px",
              }
            }
          />
        </div>
        <div
          className="self-center"
        >
          <a
            href="#"
          >
            Rules
          </a>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="max-w-6xl flex flex-col justify-center py-5 mx-auto"
  >
    <div>
      Hola
    </div>
  </div>
</div>
`;
